images/macos/software-report/SoftwareReport.Generator.ps1

```diff
@@ -22,322 +22,256 @@ Import-Module "$PSScriptRoot/../helpers/Xcode.Helpers.psm1"
-$toNatural = { [regex]::Replace($_, '\d+', { $args[0].Value.PadLeft(20) }) }
-$markdown += New-MDList -Style Unordered -Lines ($languageAndRuntimeList | Sort-Object $toNatural)
+# Software report
+$softwareReport = [SoftwareReport]::new($osInfo)
+$installedSoftware = $softwareReport.Root.AddHeaderNode("Installed Software")
+
+# Language and Runtime
+$languageAndRuntime = $installedSoftware.AddHeaderNode("Language and Runtime")
+$languageAndRuntime.AddToolNode(".NET SDK", $(Get-DotnetVersionList))
+$languageAndRuntime.AddToolNode("Bash", $(Get-BashVersion))
+$languageAndRuntime.AddNodes($(Get-ClangLLVMVersions))
+$languageAndRuntime.AddNodes($(Get-GccVersions))
+$languageAndRuntime.AddNodes($(Get-FortranVersions))
+$languageAndRuntime.AddToolNode("Go", $(Get-GoVersion))
+$languageAndRuntime.AddToolNode("Julia", $(Get-JuliaVersion))
+$languageAndRuntime.AddToolNode("Kotlin", $(Get-KotlinVersion))
+$languageAndRuntime.AddToolNode("MSBuild", $(Get-MSBuildVersion))
+$languageAndRuntime.AddToolNode("Node.js", $(Get-NodeVersion))
+$languageAndRuntime.AddToolNode("NVM", $(Get-NVMVersion))
+$languageAndRuntime.AddToolNode("NVM - Cached node versions", $(Get-NVMNodeVersionList))
+$languageAndRuntime.AddToolNode("Perl", $(Get-PerlVersion))
+$languageAndRuntime.AddToolNode("PHP", $(Get-PHPVersion))
+$languageAndRuntime.AddToolNode("Python", $(Get-PythonVersion))
+$languageAndRuntime.AddToolNode("Python3", $(Get-Python3Version))
+$languageAndRuntime.AddToolNode("R", $(Get-RVersion))
+$languageAndRuntime.AddToolNode("Ruby", $(Get-RubyVersion))
 
 # Package Management
-$markdown += New-MDHeader "Package Management" -Level 3
-$packageManagementList = @(
-    (Get-PipVersion -Version 2),
-    (Get-PipVersion -Version 3),
-    (Get-PipxVersion),
-    (Get-BundlerVersion),
-    (Get-CocoaPodsVersion),
-    (Get-CondaVersion),
-    (Get-HomebrewVersion),
-    (Get-NPMVersion),
-    (Get-YarnVersion),
-    (Get-NuGetVersion),
-    (Get-RubyGemsVersion),
-    (Get-ComposerVersion),
-    (Get-CarthageVersion),
-    (Get-VcpkgVersion)
-)
-
-$markdown += New-MDList -Style Unordered -Lines ($packageManagementList | Sort-Object)
-$markdown += New-MDHeader "Environment variables" -Level 4
-$markdown += Build-PackageManagementEnvironmentTable | New-MDTable
-$markdown += New-MDNewLine
+$packageManagement = $installedSoftware.AddHeaderNode("Package Management")
+$packageManagement.AddToolNode("Bundler", $(Get-BundlerVersion))
+$packageManagement.AddToolNode("Carthage", $(Get-CarthageVersion))
+$packageManagement.AddToolNode("CocoaPods", $(Get-CocoaPodsVersion))
+$packageManagement.AddToolNode("Composer", $(Get-ComposerVersion))
+$packageManagement.AddToolNode("Homebrew", $(Get-HomebrewVersion))
+$packageManagement.AddToolNode("Miniconda", $(Get-CondaVersion))
+$packageManagement.AddToolNode("NPM", $(Get-NPMVersion))
+$packageManagement.AddToolNode("NuGet", $(Get-NuGetVersion))
+$packageManagement.AddToolNode("Pip2", $(Get-PipVersion -Version 2))

```

Comment:
`Pip2` should be `Pip`. Sorry for confusing you during our previous discussion.
Pip for Python 2 is named `Pip`. Pip for Python 3 is named `Pip3`

```suggestion
$packageManagement.AddToolNode("Pip", $(Get-PipVersion -Version 2))

```